apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,542 @@
+    dict: A dict of the task definition and it's ARN if succesful, and an error dictionary if not
+    """
+    container_name = "worker_{}".format(queue_name)
+    image = "{AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-{env}-worker:latest".format(AWS_ACCOUNT_ID=aws_keys["AWS_ACCOUNT_ID"], env=ENV)
+    AWS_DEFAULT_REGION = aws_keys["AWS_REGION"]
+    log_group = "{}_logs".format(queue_name)
+
+    task_role_arn = TASK_ROLE_ARN
+    execution_role_arn = TASK_EXECUTION_ROLE_ARN
+
+    if task_role_arn and execution_role_arn:
+        definition = task_definition.format(queue_name=queue_name, task_role_arn=task_role_arn,
+                                            execution_role_arn=execution_role_arn, container_name=container_name,
+                                            image=image, AWS_DEFAULT_REGION=AWS_DEFAULT_REGION, env=ENV,
+                                            challenge_pk=challenge.pk, log_group=log_group, **DJANGO_SETTINGS_DICT)
+        definition = eval(definition)
+        if not challenge.task_def_arn:
+            try:
+                response = client.register_task_definition(**definition)
+                if (response["ResponseMetadata"]["HTTPStatusCode"] == HTTPStatus.OK):
+                    task_def_arn = response["taskDefinition"]["taskDefinitionArn"]
+                    challenge.task_def_arn = task_def_arn
+                    challenge.save()
+                return response
+            except ClientError as e:
+                logger.info(e)
+                return e.response
+        else:
+            message = "Error. Task definition already registered for challenge {}.".format(challenge.pk)
+            return {"Error": message, "ResponseMetadata": {"HTTPStatusCode": 400}}
+    else:
+        message = "Please ensure that the TASK_ROLE_ARN & TASK_EXECUTION_ROLE_ARN are appropriately passed as environment varibles."
+        return {"Error": message, "ResponseMetadata": {"HTTPStatusCode": 400}}
+
+
+def create_service_by_challenge_pk(client, challenge, client_token):
+    """
+    Creates the worker service for a challenge, and sets the number of workers to one.
+
+    Parameters:
+    client (boto3.client): the client used for making requests to ECS
+    challenge (<class 'challenges.models.Challenge'>): The challenge object  for whom the task definition is being registered.
+    client_token (str): The client token generated by client_token_generator()
+
+    Returns:
+    dict: The response returned by the create_service method from boto3. If unsuccesful, returns an error dictionary
+    """
+
+    queue_name = challenge.queue
+    service_name = "{}_service".format(queue_name)
+    if challenge.workers is None:  # Verify if the challenge is new (i.e, service not yet created.).
+        if challenge.task_def_arn is "":
+            response = register_task_def_by_challenge_pk(client, queue_name, challenge)
+            if (response["ResponseMetadata"]["HTTPStatusCode"] != HTTPStatus.OK):
```

Comment:
@KhalidRmb it should be `(` everywhere or no where, please make this consistent over whole codebase

Reply:
I'm mainly using them when there is more than one condition to verify, for readablity. I'll make all if statements bracketed in that case.